workers/unity/Packages/com.improbable.gdk.playerlifecycle/Systems/PlayerHeartbeat/SendPlayerHeartbeatRequestSystem.cs

```diff
@@ -23,11 +23,10 @@ public class SendPlayerHeartbeatRequestSystem : ComponentSystem
         private struct Data
         {
             public readonly int Length;
-            [ReadOnly] public ComponentDataArray<PlayerHeartbeatClient.CommandSenders.PlayerHeartbeat> RequestSenders;
+            public ComponentDataArray<PlayerHeartbeatClient.CommandSenders.PlayerHeartbeat> RequestSenders;
             [ReadOnly] public ComponentDataArray<Authoritative<PlayerHeartbeatServer.Component>> AuthorityMarkers;
+            [ReadOnly] public ComponentDataArray<HeartbeatData> HasHeartbeatData;
             [ReadOnly] public ComponentDataArray<SpatialEntityId> SpatialEntityIds;
-            public EntityArray Entities;
-            public SubtractiveComponent<AwaitingHeartbeatResponseTag> NotAwaitingHeartbeatResponse;
```

Comment:
Doesn't removing this mean that it will send commands regardless of whether or not it has a response. Also that you could send two messages before receiving either response. Which in turn means you could miss the first heartbeat, but respond to the second, but end up still registering as missing a heartbeat.

Reply:
This is what adding extra components was meant to help. And generally how I would use an ECS in this case. If we want to avoid doing that right now we could add a bit to the heartbeat data.

Reply:
I'm not sure what you mean with this? The change here means that the heartbeat will be send at a set interval, not caring about if it was responded to or not.

Reply:
Which is the race that the awaiting response tag was meant to solve

Reply:
As we also aren't cross checking response orders against send orders.
We could also solve this race by having a permanent component store the data in state instead.
Then early out if we are waiting for a response.

Reply:
I dont see how this is a race?

Reply:
I described the race in the first comment.
Server sends ping A
Server sends ping B
Client sends pong B
Client sends pong A too late (which arrived later)
Server receives pong B
Server receives pong B failure - updates heartbeats failed to 1

Reply:
Can also check against IDs. This is going to happen when overloaded although probably not much during normal operation (unless timeouts are too low). We could also not deal with this for now and fix it later but be aware this is semantically changing something that was done deliberately.
Arguably kicking clients more often when overloaded is fine, although that behaviour happens anyway. This just adds some oddity to it.